Sharper parameter estimation

gerkovink opened the pull request:
Extension of the core function `.norm.draw` with a new function `estimice` that allows for new methods to obtain the least squares parameters: (the current) ridge regression, QR decomposition (new default) and singular value decomposition. Ridge regression introduces some bias (albeit normally small), which can be bypassed with QR decomposition and SVD for  sharper parameter estimation.

stefvanbuuren commented:
Good suggestions that will improve stability of the algorithm. I have two requests: 

1. The `estimice()` function writes a lengthy message to the user console when it uses the ridge method as back-up. I'd rather inform the user in a less obtrusive way by adding a short message to the log file, for example as done in `remove.lindep`. You may need to experiment a little to find the  appropriate number of upstair frames.
2. The test file is useful. I would suggest using `context()` and `testthat()` functionality so that it conforms to other test function. Also, define the explicit warning or error message whether you expect one.

gerkovink commented:
Should be allright now. Summary of major changes:

- Added the requested `testthat` extension as well as some additional tests
- Iteration history now prints `*` if and only if ridge regression is used as a fallback (changes in `estimate` and `sampler`)
- Printing inherits from `printFlag`
- Added a replicable example for `mice.mids` (Issue #65)
- Some efficiency gains in `mice.impute.quadratic`
- Solved potential error in `mice.impute.quadratic` [Thanks Anna-Carolina Haensch]
